cmd/swarm/swarm-snapshot/verify.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2018 The go-ethereum Authors
+// This file is part of go-ethereum.
+//
+// go-ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// go-ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with go-ethereum. If not, see <http://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/node"
+	"github.com/ethereum/go-ethereum/p2p/simulations/adapters"
+	"github.com/ethereum/go-ethereum/swarm/network"
+	"github.com/ethereum/go-ethereum/swarm/network/simulation"
+	cli "gopkg.in/urfave/cli.v1"
+)
+
+// verify is used as the entry function for "verify" app command.
+func verify(ctx *cli.Context) error {
+	log.PrintOrigins(true)
+	log.Root().SetHandler(log.LvlFilterHandler(log.Lvl(ctx.Int("verbosity")), log.StreamHandler(os.Stdout, log.TerminalFormat(true))))
+
+	if len(ctx.Args()) < 1 {
+		return errors.New("argument should be the filename to verify")
+	}
+	return verifySnapshot(ctx.Args()[0])
+}
+
+// verifySnapshot constructs a simulation, uploads a snapshot from
+// a file with provided filename and validates that kademlia is healthy.
+func verifySnapshot(filename string) (err error) {
```

Comment:
Why do we need this?

Reply:
Do you say that verify function is not needed?

Reply:
No, I am asking what do we need it for :)

Reply:
Well, to validate existing snapshot, isn't it obvious?

Reply:
Lol. Yes. It is obvious.

So in what situations would we want to verify the snapshot? Shouldn't tests in simulations verify that snapshot generation is working properly?

Reply:
Maybe, should I remove it, then?

Reply:
I believe we should, yes.

Reply:
At the present time I can't think of what subcommands we would want to have here. For my part the create subcommand can be taken out. Or it can be left in and assigned as the default command if only one argument is given. I don't have a strong opinion which alternative is better.

Reply:
Cool, thanks. @zelig @justelad any opinions on subcommands? I am also fine with both approaches.

Reply:
i opt for dropping subcmd

Reply:
removed the file as there's an apparent quorum on this